### PR TITLE
[JUJU-1844] Bash-based version of nw-deploy-bionic-aks-test

### PR DESCRIPTION
**Pull request by anvial**
This PR adds a bash-based version of the nw-deploy-bionic-aks-test.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made

## QA steps

```sh
cd tests ; ./main.sh -v deploy_caas run_deploy_aks
```

### CODE REVIEW

**Review by SimonRichardson — APPROVED** [review]

**Review by jack-w-shaw — CHANGES_REQUESTED** [review]
Static analysis is failing. There's a mix of tabs and spaces

File: tests/suites/deploy_caas/task.sh [hunk 1]
```diff
@@ -11,6 +11,15 @@ test_deploy_caas() {
...
 	file="${TEST_DIR}/test-deploy-caas.log"
 
+	case "${BOOTSTRAP_PROVIDER:-}" in
+  "k8s")
+    test_deploy_aks
+    ;;
+  *)
+    echo "==> TEST SKIPPED: test_deploy_caas test runs on k8s only"
+    ;;
+  esac
```
Comment (jack-w-shaw) [inline tests/suites/deploy_caas/task.sh:21]: I don't really think this integration test belongs here

We are always running a test in azure if the provider is 'k8s'. Ideally we would treat aks as it's own provider. This is particularly nasty given we bootstrap a controller inside of the unit test itself.

Creating a new aks provider will become necessary if we want any more integration tests, but it'll be a fairly substantial bit of work. My preference in the short term would be to extract this out into its own integration test. It's not nice that we have to bootstrap a controller in aks, successfully run the integration test, then tear down before we even start `test_desploy_charm` in it's own controller. Ideally each integration test suite would only require bootstrapping a single controller

File: tests/includes/controller.sh [hunk 2]
```diff
@@ -16,3 +16,23 @@ bootstrap_alt_controller() {
 	END_TIME=$(date +%s)
 	echo "====> Bootstrapped ${name} ($((END_TIME - START_TIME))s)"
 }
+
+# Bootstraps custom controller in test
+bootstrap_custom_controller() {
+	local name, cloud_name
+
+	name=${1}
+  cloud_name=${2}
```
Comment (jack-w-shaw) [inline tests/includes/controller.sh:25]: Mixing spaces + tabs

File: tests/suites/deploy_aks/task.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	case "${BOOTSTRAP_PROVIDER:-}" in
+	"k8s")
+		test_deploy_aks
+		;;
+	*)
+		echo "==> TEST SKIPPED: test_deploy_caas test runs on k8s only"
+		;;
+	esac
+
+	juju destroy-controller aks --destroy-all-models --destroy-storage -y
```
Comment (jack-w-shaw) [inline tests/suites/deploy_aks/task.sh:27]: The `includes` helper `destroy_controller` should probably be used here

File: tests/suites/deploy_aks/deploy_aks.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+	wait_for "postgresql-k8s" "$(idle_condition "postgresql-k8s" 1)"
+	wait_for "mattermost-k8s" "$(idle_condition "mattermost-k8s" 0)"
+
+	echo "Verify application is reachable"
+	mattermost_ip="$(juju status --format json | jq -r '.applications["mattermost-k8s"].units["mattermost-k8s/0"].address')"
+	juju run --unit mattermost-k8s/0 "curl ${mattermost_ip}:8065 >/dev/null"
+
+	echo "Destroy model"
+	juju destroy-model "test-deploy-aks" -y
```
Comment (jack-w-shaw) [inline tests/suites/deploy_aks/deploy_aks.sh:22]: The `includes` helper `destroy_model` should probably be used here

**Review by jack-w-shaw — APPROVED** [review]
Approved, once some minor fixes have been made

**Comment by jack-w-shaw** [comment]
Also QA steps need to be fixed

**Review by jack-w-shaw — CHANGES_REQUESTED** [review]
Have just realised deploy_aks needs to be added to main.sh

File: tests/suites/deploy_aks/task.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+test_deploy_caas() {
```
Comment (jack-w-shaw) [inline tests/suites/deploy_aks/task.sh:1]: This should be test_deploy_aks

File: tests/suites/deploy_aks/deploy_aks.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+test_deploy_aks() {
```
Comment (jack-w-shaw) [inline tests/suites/deploy_aks/deploy_aks.sh:25]: This is being executted when you run `./main.sh deploy_aks` without running anything in `task.sh`, so a result the bootstrap is skipped. This should be `run_*` or smth

**Review by jack-w-shaw — APPROVED** [review]

**Pull request merged**